bin/trigger_changelog.py

```diff
@@ -0,0 +1,181 @@
+    parser.add_argument('--end_date', '-ed', type=str, required=False, help=help)
+    args = parser.parse_args()
+    Constants.GIT_USERNAME = args.username
+    Constants.GIT_TOKEN = args.token
+    return args.start_date, args.end_date, args.repo, args.branch
+
+
+def main():
+    start_date, end_date, repo_name, branch = gather_args()
+    print("Gathering all repos used within the image.\nThis script may take awhile.\n")
+    repos_dict = recursive_get_repos_from_rosinstall(
+        'shadow-robot', repo_name, branch, {})
+    repo_prs = {}
+    print("Gathering all pr's within timeframe.\n")
+    for repo in repos_dict:
+        repo_data = get_prs_since_date(repo, start_date, end_date)
+        if repo_data:
+            repo_prs[repo] = repo_data
+    format_message(repo_prs)
+
+
+def get_prs_since_date(repo, start_date, end_date):
+    """ Format of data given from API Call.
+    {
+    "sr-ros-interface": {},
+    "sr_common": {
+        "https: //github.com/shadow-robot/sr_common/pull/126": [
+            "F refactoring sr description multi",
+            "F_refactoring_sr_description_multi",
+            "n/a",
+            "2022-01-03T15:59:51Z"
+        ],
+        "https://github.com/shadow-robot/sr_common/pull/127": [
+            "adding DIRECT_PWM_MODE to teach_mode_node",
+            "SRC-6613_update_controller_guis",
+            "https://shadowrobot.atlassian.net/browse/SRC-6613",
+            "2022-01-18T12:17:08Z"
+        ]
+      }
+    }
+    """
+    if repo in Constants.EXCLUDED_REPOS:
+        return None
+
+    if end_date:
+        query_url = Constants.GITHUB_API_URL+"/search/issues?q=repo:shadow-robot/" + \
+            f"{repo}+is:pr+is:merged+sort:updated-asc+merged:{start_date}..{end_date}"
+    else:
+        query_url = Constants.GITHUB_API_URL+"/search/issues?q=repo:shadow-robot/" + \
+            f"{repo}+is:pr+is:merged+sort:updated-asc+merged:>{start_date}"
+        
+    time.sleep(2)
+    query_result = requests.get(query_url, auth=(
+        Constants.GIT_USERNAME, Constants.GIT_TOKEN))
+    result_json = json.loads(query_result.text)
+    items = result_json['items']
+    prs = {}
+    for item in items:
+        pr_number = str(item['number'])
+        pr_url = item['pull_request']['html_url']
+        pr_title = item['title']
+        pr_query = Constants.GITHUB_API_URL + \
+            "/repos/shadow-robot/"+repo+"/pulls/"+pr_number
+        pr_query_result = requests.get(pr_query, auth=(
+            Constants.GIT_USERNAME, Constants.GIT_TOKEN))
+        pr = json.loads(pr_query_result.text)
+        pr_branch = pr['head']['ref']
+        pr_body = pr['body']
+        pr_title_continued = ""
+        try:
+            if type(pr_body) == str or type(pr_body) == bytes:
+                pr_title_continued = re.search(
+                    Constants.REGEXP_TITLE_CONTINUED, pr_body).group(1)
+        except AttributeError:
+            continue
```

Comment:
Although you don't want to stop the program, printing a message that explains that something unexpected happen is always a good thing

Reply:
Not in the case of this script, as you want to be able to copy the whole log and paste it into the change logs, if these conditionals fail then its fine as it means the value did not exist in the API call, and that's okay behaviour